src/Vector3_TEST.cc

```diff
@@ -146,6 +146,22 @@ TEST(Vector3dTest, Vector3d)
   EXPECT_EQ(math::Vector3i::UnitZ, math::Vector3i(0, 0, 1));
 }
 
+/////////////////////////////////////////////////
+TEST(Vector3dTest, Iterator)
```

Comment:
Could you add some more tests to check if the iterator is robust to different situations? For example, change the value of the vector while holding an iterator and check how the values change. Compare against `end()`... Also add unit tests for all `Iterator` functions.